test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -58,6 +58,70 @@ test:like(output, prompt .. '^C\n---\n...\n\n' .. prompt,
 ph:shutdown({stdin = true})
 ph:close()
 
+--
+-- gh-7109: Ctrl+C does not break multiline input.
+--
+cmd = 'ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1'
+ph = popen.new({cmd}, {
+    shell = true,
+    setsid = true,
+    group_signal = true,
+    stdout = popen.opts.PIPE,
+    stderr = popen.opts.DEVNULL,
+    stdin = popen.opts.PIPE,
+})
+assert(ph, 'process is not up')
+
+start_time = clock.monotonic()
+time_quota = 10.0
+output = ''
+local prompt_name = 'tarantool'
+local expected_output = prompt
+while output:find(expected_output) == nil
+        and clock.monotonic() - start_time < time_quota do
+    output = output .. ph:read({timeout = 1.0})
+end
+assert(clock.monotonic() - start_time < time_quota, 'time_quota is violated')
+
+ph:write('box.cfg{\n')
+
+start_time = clock.monotonic()
+time_quota = 10.0
+expected_output = expected_output .. 'box.cfg{' .. '\n' ..
+        string.rep(' ', #prompt_name) .. '> '
```

Comment:
Nit: `function foo()` would have less side effects comparing to `box.cfg{`, but allows to test the same. Or, say, `foo(`.